engine/wwtlib/ScriptInterface.cs

```diff
@@ -320,65 +320,93 @@ public ImageSetLayer LoadFits(string url)
-
-            FitsImage img = new FitsImage(url, null, delegate (WcsImage wcsImage)
-            {
-                int width = (int)wcsImage.SizeX;
-                int height = (int)wcsImage.SizeY;
-
-                Imageset imageset = Imageset.Create(
-                            wcsImage.Description,
-                            Util.GetHashCode(wcsImage.Filename).ToString(),
-                            ImageSetType.Sky,
-                            BandPass.Visible,
-                            ProjectionType.SkyImage,
-                            Util.GetHashCode(wcsImage.Filename),
-                            0,
-                            0,
-                            256,
-                            wcsImage.ScaleY,
-                            ".tif",
-                            wcsImage.ScaleX > 0,
-                            "",
-                            wcsImage.CenterX,
-                            wcsImage.CenterY,
-                            wcsImage.Rotation,
-                            false,
-                            "",
-                            false,
-                            false,
-                            1,
-                            wcsImage.ReferenceX,
-                            wcsImage.ReferenceY,
-                            wcsImage.Copyright,
-                            wcsImage.CreditsUrl,
-                            "",
-                            "",
-                            0,
-                            ""
-                            );
-
-                imageset.WcsImage = wcsImage;
-                imagesetLayer.ImageSet = imageset;
-                LayerManager.AddFitsImageSetLayer(imagesetLayer, name);
-                LayerManager.LoadTree();
+            return AddImageSetLayer(url, name, gotoTarget, loaded);
+        }
+
+        public ImageSetLayer AddImageSetLayer(string url, string name, bool gotoTarget, ImagesetLoaded loaded)
+        {
+            Imageset imageset = WWTControl.Singleton.GetImageSetByUrl(url);
+            if (imageset != null)
+            {
+                if (string.IsNullOrWhiteSpace(name))
+                {
+                    name = LayerManager.GetNextImageSetName();
+                }
+                ImageSetLayer imagesetLayer = LayerManager.AddImageSetLayer(imageset, name);
+
                 if (gotoTarget)
                 {
-                    WWTControl.Singleton.GotoRADecZoom(wcsImage.CenterX / 15, wcsImage.CenterY, 10 * wcsImage.ScaleY * height, false, null);
+                    WWTControl.Singleton.GotoRADecZoom(imageset.CenterX / 15, imageset.CenterY, 
+                        WWTControl.Singleton.RenderContext.ViewCamera.Zoom, false, null);
                 }
                 if (loaded != null)
                 {
                     loaded(imagesetLayer);
                 }
-            });
 
-            return imagesetLayer;
+                return imagesetLayer;
+            } else
```

Comment:
Connecting to what I wrote above, I don't feel like we want to have behavior here where everything that isn't a known imageset is treated as a FITS file. I think it will be very useful to have a generic "turn URL into an imageset layer" API but it needs to be future-proof for new data types.